src/bentoml/_internal/runner/runner_handle/remote.py

```diff
@@ -274,6 +274,71 @@ async def async_run_method(
 
         return AutoContainer.from_payload(payload)
 
+    def async_stream_method(
+        self,
+        __bentoml_method: RunnerMethod[t.Any, P, R],
+        *args: P.args,
+        **kwargs: P.kwargs,
+    ) -> t.AsyncGenerator[R, None]:
+        import aiohttp
+
+        from ...runner.container import AutoContainer
+
+        inp_batch_dim = __bentoml_method.config.batch_dim[0]
+
+        headers = {
+            "Bento-Name": component_context.bento_name,
+            "Bento-Version": component_context.bento_version,
+            "Runner-Name": self._runner.name,
+            "Yatai-Bento-Deployment-Name": component_context.yatai_bento_deployment_name,
+            "Yatai-Bento-Deployment-Namespace": component_context.yatai_bento_deployment_namespace,
+        }
+        total_args_num = len(args) + len(kwargs)
+        headers["Args-Number"] = str(total_args_num)
+
+        if total_args_num == 1:
+            # FIXME: also considering kwargs
+            if len(kwargs) == 1:
+                kwarg_name = list(kwargs.keys())[0]
+                headers["Kwarg-Name"] = kwarg_name
+                payload = AutoContainer.to_payload(
+                    kwargs[kwarg_name], batch_dim=inp_batch_dim
+                )
+            else:
+                payload = AutoContainer.to_payload(args[0], batch_dim=inp_batch_dim)
+            data = payload.data
+
+            headers["Payload-Meta"] = json.dumps(payload.meta)
+            headers["Payload-Container"] = payload.container
+            headers["Batch-Size"] = str(payload.batch_size)
+
+        else:
+            payload_params = Params[Payload](*args, **kwargs).map(
+                functools.partial(AutoContainer.to_payload, batch_dim=inp_batch_dim)
+            )
+            data = pickle.dumps(payload_params)  # FIXME: pickle inside pickle
+
+        path = "" if __bentoml_method.name == "__call__" else __bentoml_method.name
+
+        async def stream_response() -> t.AsyncGenerator[R, None]:
+            """
+            Async generator that yields results from the remote runner.
+            """
+            try:
+                async with self._client.post(
+                    f"{self._addr}/{path}",
+                    data=data,
+                    headers=headers,
+                ) as resp:
+                    async for line in resp.content.iter_any():
```

Comment:
Are we sure we want to use newline as a separator? Can pickled content not contain a newline?

Reply:
I don't think we are explicitly using newline as separator, starlette handles the chunking delimiter when you pass a generator to the `StreamingResponse`. on the client here we are essentially iterating the chunks 
(i could be wrong)

should probably use `chunk` instead of `line` here

Reply:
actually you might be right

Reply:
I think chunking should be transparent to us, yeah.

Reply:
what would you suggest here?

Reply:
We need to have our own separator if we want separate messages in a stream.

Reply:
Actually, given pickle probably contains arbitrary binary data, we'd probably have to do something like
`<content length> <pickle content>`
and use that to get each object off the stream.

Reply:
actually after looking up a little, what I initially said was probably right. `<content length> <pickle content>` is actually being abstracted away by `StreamingResponse` and `iter_any`. Actually most ASGI clients will handle the chunking for us in this form https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding#chunked_encoding

I think we can keep this at it is until we see the need of a lower level control?

Reply:
As long as we can ensure both the server and client are sending chunk boundaries and not truncating chunks at a max size, that works, though I think you want `iter_chunks`.

Reply:
My test with `iter_any` is that it will dynamically read the chunks without a fixed bytes length, though not sure why its a much lesser documented API

Reply:
My understanding of the [aiohttp documentation](https://docs.aiohttp.org/en/stable/streams.html#asynchronous-iteration-support) that `iter_any` will give you partial chunks if the chunks are very large.

Have you verified that very large chunks on the server side are not split into smaller chunks?

Reply:
To iterate by line, simply do `async for line in resp.content`
https://docs.aiohttp.org/en/stable/streams.html#asynchronous-iteration-support

Reply:
We don't want to iterate by line as pickle content may include newlines; we must either use chunk boundaries or implement our own chunking, essentially.

Reply:
Thanks @sauyon for pointing out this. Tested on a big chunk of data with a single `yield` on the server side, indeed it will be split into multiple chunks. I added logic to handle this